stackstate_checks_base/tests/test_http_helper.py

```diff
@@ -0,0 +1,458 @@
+
+    def test_headers(self):
+        assert HTTPHelperRequestHandler().validate_headers() is True
+        assert HTTPHelperRequestHandler({"headers": {"a": "1"}}).get_request_model().headers == {"a": "1"}
+        assert HTTPHelperRequestHandler({"headers": {}}).get_request_model().headers == {}
+        assert HTTPHelperRequestHandler({"headers": {"a": "1"}}).validate_headers() is True
+        # There is no fail clause as you are unable to pass anything else than a dict or the schematic will complain
+
+    def test_authentication(self):
+        auth_valid = HTTPHelperRequestHandler({
+            "auth_type": HTTPAuthenticationType.BasicAuth,
+            "auth_data": {"username": "test", "password": "test"}
+        })
+        assert auth_valid.get_request_model().auth_data == {"username": "test", "password": "test"}
+        assert auth_valid.get_request_model().auth_type == HTTPAuthenticationType.BasicAuth
+
+        auth_invalid = HTTPHelperRequestHandler({
+            "auth_type": -1,
+            "auth_data": {}
+        })
+        assert auth_invalid.get_request_model().auth_data == {}
+        assert auth_invalid.get_request_model().auth_type == -1
+
+        assert auth_valid.validate_auth() is True
+        try:
+            assert auth_invalid.validate_auth() is True
+            assert False
+        except TypeError:
+            assert True
+
+    def test_body_type_validation(self):
+        assert HTTPHelperRequestHandler().validate_body_type() is True
+        assert HTTPHelperRequestHandler({"request_type_validation": HTTPRequestType.JSON})\
+               .get_request_model().request_type_validation == HTTPRequestType.JSON
+        assert HTTPHelperRequestHandler({"request_type_validation": -1}) \
+               .get_request_model().request_type_validation == -1
+        assert HTTPHelperRequestHandler({
+            "body": {},
+            "request_type_validation": HTTPRequestType.JSON
+        }).validate_body_type()
+        try:
+            assert HTTPHelperRequestHandler({
+                "request_type_validation": HTTPRequestType.JSON
+            }).validate_body_type()
+            assert False
+        except ValueError:
+            assert True
+
+    def test_body_schematic_validation(self):
+        assert HTTPHelperRequestHandler().validate_body_schematic() is True
+        assert HTTPHelperRequestHandler({"request_schematic_validation": BodyRequestSchematicTest}) \
+               .get_request_model().request_schematic_validation == BodyRequestSchematicTest
+        assert HTTPHelperRequestHandler({"request_schematic_validation": -1}) \
+               .get_request_model().request_schematic_validation == -1
+        assert HTTPHelperRequestHandler({
+            "body": {
+                "hello": "world",
+                "pong": True
+            },
+            "request_schematic_validation": BodyRequestSchematicTest
+        }).validate_body_schematic()
+        try:
```

Comment:
Typically unit test frameworks allow you to test for the presence of exceptions. 

pytest is no exception: https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions 

pun intended :)

Reply:
Replaced with pytest